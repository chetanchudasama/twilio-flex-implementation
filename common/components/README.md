# Common Components Library

The common components library contains UI elements and models that can be used across multiple Flex plugins.

It is written in react v16.13.1 and typescript v3.

## Contents

## Pre-requisites

See Pre-requisites section in the [main README.md file](../../README.md).

## Building the library

See Scripts section in the [main README.md file](../../README.md) for building and running the unit tests for this library.
When building the library, the components contained within will be compiled to CommonJS and ESM formats and a `lib` folder will be created containing the compiled files. The Flex plugins will then be able to reference any of the components in the library.
