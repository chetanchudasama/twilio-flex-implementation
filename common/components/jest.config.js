module.exports = {
	watchPlugins: [
		"jest-watch-typeahead/filename",
		"jest-watch-typeahead/testname",
	],
	verbose: true,
	moduleFileExtensions: ["js", "jsx", "ts", "tsx"],
	moduleDirectories: ["node_modules", "src"],
	moduleNameMapper: {
		"\\.(css|less|scss)$": "identity-obj-proxy",
	},
	modulePathIgnorePatterns: ["<rootDir>/lib/"],
	transform: {
		"^.+\\.(ts|tsx)?$": "ts-jest",
	},
	globals: {
		window: true,
		navigator: {
			userAgent: "node",
		},
	},
	coveragePathIgnorePatterns: ["node_modules", ".min.js", "tests", "utils"],
	setupFilesAfterEnv: ["<rootDir>/src/setupTests.ts"],
};
