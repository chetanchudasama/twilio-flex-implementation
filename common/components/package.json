{
  "name": "@common/components",
  "version": "0.1.0",
  "main": "lib/cjs/index.js",
  "typings": "lib/cjs/index.d.ts",
  "module": "lib/esm/index.js",
  "scripts": {
    "test": "jest"
  },
  "dependencies": {
    "@date-io/date-fns": "^1.3.13",
    "@material-ui/core": "^3.9.3",
    "@material-ui/icons": "^3.0.2",
    "@material-ui/lab": "^3.0.0-alpha.30",
    "material-ui-flat-pagination": "^3.2.1",
    "material-ui-pickers": "^2.2.4",
    "moment": "^2.29.1",
    "react": "16.13.1",
    "react-dom": "16.13.1",
    "react-image-lightbox": "^5.1.1",
    "react-emotion": "^9.2.6",
    "date-fns": "^2.22.1"
  },
  "devDependencies": {
    "@types/enzyme": "^3.10.8",
    "@types/enzyme-adapter-react-16": "^1.0.6",
    "@types/jest": "^26.0.23",
    "@types/react": "16.9.56",
    "@types/react-test-renderer": "16.9.5",
    "enzyme": "^3.11.0",
    "enzyme-adapter-react-16": "1.15.6",
    "flush-promises": "^1.0.2",
    "jest": "^26.6.3",
    "react-test-renderer": "16.13.1",
    "ts-jest": "^26.5.6"
  },
  "peerDependencies": {
    "react": "16.13.1",
    "react-dom": "16.13.1"
  }
}
