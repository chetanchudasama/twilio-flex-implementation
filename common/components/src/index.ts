export * from "./CustomCRM/CustomCRM";
export * from "./CustomerBanner/CustomerBanner";
export * from "./CustomerBannerMenu/CustomerBannerMenu";
export * from "./CustomerDetailActionWrapper/CustomerDetailActionWrapper";
export * from "./CustomInput/CustomInput";
export * from "./CustomMultiSelect/CustomMultiSelect";
export * from "./CustomSelect/CustomSelect";
export * from "./CustomSnackbar/CustomSnackbar";
export * from "./DocumentDownload/DocumentDownload";
export * from "./Icons/Icons";
export * from "./ImageCarousel/ImageCarousel";
export * from "./ImagePreview/ImagePreview";
export * from "./models/AddressItemModel";
export * from "./models/ApplicationStatusModel";
export * from "./models/BankDetailModel";
export * from "./models/BaseAddressModel";
export * from "./models/BaseWizardStepProps";
export * from "./models/CustomerDetailModel";
export * from "./models/CustomerEmploymentModel";
export * from "./models/ItemModel";
export * from "./models/OccupationStatusModel";
export * from "./models/PreferredLenderModel";
export * from "./models/ThirdPartyAuthorizationDetailModel";
export * from "./models/VehicleMakeData";
export * from "./models/VehicleModelData";
export * from "./models/VulnerabilityReason";
export * from "./models/VulnerableCustomerInformation";
export * from "./MuiDatePicker/MuiDatePicker";
export * from "./Pagination/Pagination";
export * from "./ResponsiveDialog/index";
export * from "./services/base.service";
export * from "./shared/AppTheme";
export * from "./shared/enum";
export * from "./shared/Shared";
export * from "./SnackbarMessage/SnackbarMessage";
export * from "./StyledButtons/StyledLinkButton";
export * from "./ThreeStateToggle/ThreeStateToggle";
export * from "./Wizard/Wizard";
export * from "./WizardStepWrapper/WizardStepWrapper";
export * from "./models/ResidentialStatusModel";
export * from "./models/TitleModel";
export * from "./models/MaritalStatusModel";
export * from "./models/DrivingLicenceTypeModel";
export * from "./CallbackDialog/CallbackDialog";
