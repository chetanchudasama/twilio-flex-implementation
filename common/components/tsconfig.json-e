{
  "extends": "../../tsconfig.json",
  "compilerOptions": {
    "jsx": "react",
    "module": "esnext",
    "esModuleInterop": true,
    "outDir": "lib/esm",
    "rootDir": "src",
    "lib": ["ES2018", "DOM"],
    "allowSyntheticDefaultImports": true
  },
  "exclude": ["node_modules", "lib/**/*"],
  "references": []
}
