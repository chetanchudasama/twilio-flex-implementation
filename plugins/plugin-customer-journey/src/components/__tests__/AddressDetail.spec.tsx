import React from "react";
import { create } from "react-test-renderer";
import { mount } from "enzyme";
import { Shared } from "@common/components";
import {
  AddressDetail,
  AddressItemProps,
} from "../CustomerDetailAction/AddressHistory/AddressDetail/AddressDetail";

const props: AddressItemProps = {
  addressItem: {
    postcode: "NN6 2XS",
    buildingName: "Grange",
    subBuilding: "Grange-1",
    buildingNumber: "110",
    streetName: "43 Buckingham Rd",
    town: "THORNBY",
    addressStatus: {
      addressStatusId: 1,
      addressStatusName: "Home Owner",
      yearsAtAddress: 1,
      monthsAtAddress: 12,
    },
    isPrimaryAddress: true,
  },
  onEdit: jest.fn(() => {}),
};

describe("AddressDetail", () => {
  it("renders correctly", () => {
    const wrapper = create(<AddressDetail {...props} />).toJSON();

    expect(wrapper).toMatchSnapshot();
  });

  it("should render address value correctly in UI", () => {
    const wrapper = create(<AddressDetail {...props} />).root;

    expect(
      wrapper.findByProps({
        className: "address-content address-line1",
      }).props.children
    ).toContain(
      `${props.addressItem.buildingNumber}, ${props.addressItem.buildingName}, ${props.addressItem.subBuilding}, ${props.addressItem.streetName}`
    );
    expect(
      wrapper.findByProps({
        className: "address-content address-line2",
      }).props.children
    ).toContain(`${props.addressItem.town}, ${props.addressItem.postcode}`);
  });

  it("when edit button is clicked, onEdit method should be emitted", () => {
    const wrapper = mount(<AddressDetail {...props} />);

    const button = wrapper.find(".edit-address-btn").first();
    button.simulate("click");
    wrapper.update();

    expect(props.onEdit).toHaveBeenCalledTimes(1);
  });
});
