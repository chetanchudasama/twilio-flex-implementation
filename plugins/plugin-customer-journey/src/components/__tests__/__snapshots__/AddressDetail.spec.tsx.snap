// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddressDetail renders correctly 1`] = `
<div
  className="MuiGrid-item-2 MuiGrid-grid-xs-12-41 css-5st4sl"
>
  <div
    className="MuiCardHeader-root-98 card-header"
  >
    <div
      className="MuiCardHeader-avatar-99"
    >
      <div
        className="MuiAvatar-root-104 MuiAvatar-colorDefault-105 icon"
        color="secondary"
      >
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root-107 MuiSvgIcon-colorSecondary-109"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
          />
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
        </svg>
      </div>
    </div>
    <div
      className="MuiCardHeader-content-101"
    >
      <span
        className="MuiTypography-root-116 MuiTypography-body2-124 MuiCardHeader-title-102"
      >
        <span
          className="MuiTypography-root-116 MuiTypography-body1-125 address-content address-line1"
        >
          110, Grange, Grange-1, 43 Buckingham Rd
        </span>
        <span
          className="MuiTypography-root-116 MuiTypography-body1-125 address-content address-line2"
        >
          THORNBY, NN6 2XS
        </span>
      </span>
      <span
        className="MuiTypography-root-116 MuiTypography-body2-124 MuiTypography-colorTextSecondary-149 MuiCardHeader-subheader-103 subheader"
      >
        <span
          className="MuiTypography-root-116 MuiTypography-body1-125 subheader-line"
        >
          Home Owner
        </span>
        <span
          className="MuiTypography-root-116 MuiTypography-body1-125 subheader-line"
        >
          1 years, 
            12 months
        </span>
      </span>
    </div>
    <div
      className="MuiCardHeader-action-100"
    >
      <button
        className="MuiButtonBase-root-178 MuiButton-root-152 MuiButton-text-154 MuiButton-textSecondary-156 MuiButton-flat-157 MuiButton-flatSecondary-159 MuiButton-sizeSmall-175 edit-address-btn"
        disabled={false}
        onBlur={[Function]}
        onClick={[MockFunction]}
        onContextMenu={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="0"
        type="button"
      >
        <span
          className="MuiButton-label-153"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-107 MuiSvgIcon-fontSizeSmall-114 edit-icon"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 0 0-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
            />
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
          </svg>
          Edit
        </span>
        <span
          className="MuiTouchRipple-root-186"
        />
      </button>
    </div>
  </div>
  <hr
    className="MuiDivider-root-181 divider"
  />
</div>
`;
